great_expectations/expectations/core/expect_column_values_to_match_regex.py

```diff
@@ -202,3 +204,61 @@ def _prescriptive_renderer(
                 }
             )
         ]
+
+    examples = [
```

Comment:
This obviously works, but what I felt we are lacking currently in the CFE is that the meaning of class-level variables is not clear until one digs further in the code.  We declare various class variables, but their semantics are not immediately clear.  I had to search further down in another module to see how the `examples` list is used.

Reply:
I agree @alexsherstinsky  -> I do not think this should be a class variable; instead these test cases should:

1. Be updated to add a key called `render_to_gallery` in the test definition to emphasize that they're intended to be visible descriptive tests;
2. Be moved outside of the class.

Reply:
As I started to implement (2) the changes to move `examples` out of the `Expectation` class, I talked myself back into my original position, but for different reasons.

I agree with James that `examples` may eventually include large asset files. And we definitely don’t want to force loading of those assets when any given `Expectation` class is loaded.

However, if we require `Expectations` to introspect their parent modules for example sets, then we tightly bind the operation of the Expectation class to the file structure that Expectations are declared in. I ran into this almost immediately when I started implementing the change in `tests/expectations/test_self_check.py`, since that file includes three sample Expectations, each with different example sets. To accommodate module-based introspection, I would have needed to rejigger a bunch of files—totally doable, but very unusual. Off the top of my head, I don’t know of any examples of python classes that insist on being stored one-to-a-file, or that introspect their parent file in this way. The more I think about it, this feels like we're trying to do something that's fundamentally at odds with the way python usually does namespaces.

Also, I realized that moving the examples to the parent module doesn’t actually solve the problem of loading large asset files. Whenever a user imports the module, the `examples =`  declaration will be run. This happens even if the user invokes `from some_expectation_module import ExpectSomething`.

So, instead of pulling `examples` out of the `Expectation` class, I vote we solve the problem directly, by implementing methods for lazy loading within the class.

In the meantime, I've implemented part (1) on a branch. Will merge it shortly.